doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+Field :  
+      "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
+      "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MOO
+DE_HW",
+      "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:<port_id>",
+      "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
+``` 
+
+### 3.3.5 COUNTER DB
+None
+
+## 3.4 Kernel Iptables
+kernel has the knowledge about how to forward multicast data traffic. So, whenever multicast data packets are lifted to CPU, the Kernel shouldn't forward them. The data traffic would have already been forwarded by MFIB entries in NPU. Drop rule will be added for each VRF in iptables FORWARD chain to refrain kernel forwarding multicast packets. The multicast data in the Register packets that are sent by FHR alone would be forwarded by the Kernel in RP. This forwarding is essential to avoid loss of traffic till the multicast (S,G) path is established. So, other than this none of the multicast forwarding is allowed by the Kernel.
+
+## 3.5 User Interface
+### 3.5.1 Data Models
+
+The existing openconfig-pim.yang and sonic-pim yangs are 
+updated with with PIM Rendezvous point and SPT related configuration.Static RP mapping and assert related show commands are added in Sonic-pim.yang 
+
+```diff
+       |     +--rw pim
+        |     |  +--rw global
+        |     |  |  +--rw config
+        |     |  |  |  +--rw join-prune-interval?      uint16
+        |     |  |  |  +--rw keep-alive-timer?         uint16
+        |     |  |  |  +--rw ecmp-enabled?             boolean
+        |     |  |  |  +--rw ecmp-rebalance-enabled?   boolean
+        |     |  |  +--ro state
+        |     |  |  |  +--ro join-prune-interval?      uint16
+        |     |  |  |  +--ro keep-alive-timer?         uint16
+        |     |  |  |  +--ro ecmp-enabled?             boolean
+        |     |  |  |  +--ro ecmp-rebalance-enabled?   boolean
+        |     |  |  +--rw ssm
+        |     |  |  |  +--rw config
+        |     |  |  |  |  +--rw ssm-ranges?   -> /oc-rpol:routing-policy/defined-sets/prefix-sets/prefix-set/name
+        |     |  |  |  +--ro state
+        |     |  |  |     +--ro ssm-ranges?   -> /oc-rpol:routing-policy/defined-sets/prefix-sets/prefix-set/name
++       |     |  |  +--rw rendezvous-points
++       |     |  |  |  +--rw rendezvous-point* [address]
++       |     |  |  |     +--rw address    -> ../config/address
++       |     |  |  |     +--rw config
++       |     |  |  |     |  +--rw address?            inet:ip-address
++       |     |  |  |     |  +--rw multicast-groups?   -> /oc-rpol:routing-policy/defined-sets/prefix-sets/prefix-set/name
++       |     |  |  |     +--ro state
++       |     |  |  |        +--ro address?            inet:ip-address
++       |     |  |  |        +--ro multicast-groups?   -> /oc-rpol:routing-policy/defined-sets/prefix-sets/prefix-set/name
++       |     |  |  +--rw spt-switchover
++       |     |  |  |  +--rw config
++       |     |  |  |  |  +--rw spt-action?                 enumeration
```

Comment:
Hope this attr is mandatory

Reply:
Yes, it is. By default spt switchover is enabled (Default Val: PIM_SPT_IMMEDIATE). In order to disable spt switchover on all or a specific multicast group range, spt-action field must be set. 

When configured via REST, spt-action must be set to PIM_SPT_INFINITY before configuring spt-infinity-prefix-list where as CLI implemetation takes care of the same internally